metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -255,6 +256,44 @@ def get_tableau_auth(
             )
         return authentication
 
+    def check_user_role(
+        self, logged_in_user: LoggedInUser
+    ) -> Dict[Union[SourceCapability, str], CapabilityReport]:
+        capability_dict: Dict[Union[SourceCapability, str], CapabilityReport] = {
+            c.SITE_PERMISSION: CapabilityReport(
+                capable=True,
+            )
+        }
+
+        failure_reason: str = (
```

Comment:
this message should include details of what role the user does actually have as well

Reply:
Added role in failure_reason at line number 278.